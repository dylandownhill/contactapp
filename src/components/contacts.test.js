import React from 'react';
import { shallow } from 'enzyme';
import Contacts from './contacts';
import TestRenderer from 'react-test-renderer'; // ES6

const testContact = {firstName:"MyFirst", lastName:"MyLast", phone:"555-123-4567",addressLine1:"Line 1",state:"AZ",}
const testContact2 = {firstName:"22222", lastName:"sssss", phone:"aaaaa",};

describe('Contacts', () => {
	

	it('renders without crashing', () => {
		  shallow(<Contacts />);
	});

	it('Contacts.handleSave', () => {

		const wrapper = shallow(<Contacts />);
		// manually trigger the callback
		wrapper.instance().handleSave( false, testContact ) ;
		wrapper.instance().handleSave( false, testContact ) ;
		wrapper.instance().handleSave( false, testContact ) ;
		expect(wrapper.instance().state.contacts.length).toEqual(3);
		wrapper.instance().handleSave( 0, testContact ) ;
		wrapper.instance().handleSave( 1, testContact ) ;
		wrapper.instance().handleSave( 2, testContact ) ;
		// This one doesn't exist so shouldn't update
		wrapper.instance().handleSave( 3, testContact ) ;
		expect(wrapper.instance().state.contacts.length).toEqual(3);
	});
	

	it('Contacts.handleOpen', () => {
		const wrapper = shallow(<Contacts />);
		wrapper.instance().handleSave( false, testContact ) ;
		wrapper.instance().handleSave( false, testContact ) ;
		wrapper.instance().handleSave( false, testContact ) ;
		
		wrapper.instance().handleOpen( 1 ) ;
		expect(wrapper.instance().state.currentlySelected).toEqual(1);
		wrapper.instance().handleOpen( 2 ) ;
		expect(wrapper.instance().state.currentlySelected).toEqual(2);
		wrapper.instance().handleOpen( 23 ) ;
		expect(wrapper.instance().state.currentlySelected).toEqual( false );
	});
	
	it('Contacts.handleDelete', () => {
		const wrapper = shallow(<Contacts />);
		wrapper.instance().handleSave( false, testContact ) ;
		wrapper.instance().handleSave( false, testContact ) ;
		wrapper.instance().handleSave( false, testContact ) ;
		
		wrapper.instance().handleDelete( 1 ) ;
		expect(wrapper.instance().state.contacts.length).toEqual(2);
		wrapper.instance().handleDelete( 1 ) ;
		expect(wrapper.instance().state.contacts.length).toEqual(1);
		wrapper.instance().handleDelete( 23 ) ;
		expect(wrapper.instance().state.contacts.length).toEqual( 1 );
		wrapper.instance().handleDelete( 0 ) ;
		expect(wrapper.instance().state.contacts.length).toEqual(0);
		wrapper.instance().handleDelete( 0 ) ;
		expect(wrapper.instance().state.contacts.length).toEqual(0);
	});
});

test('app runs', () => {
	  const testRenderer = TestRenderer.create(
			    <Contacts />,
			  );
	  let tree = testRenderer.toJSON();
	  expect(tree).toMatchSnapshot();

	  const testInstance = testRenderer.getInstance();
	  testInstance.handleSave( false, testContact ) ;
	  expect( testRenderer.toJSON() ).toMatchSnapshot();
	  testInstance.handleSave( false, testContact2 ) ;
	  expect( testRenderer.toJSON() ).toMatchSnapshot();
	  
	});