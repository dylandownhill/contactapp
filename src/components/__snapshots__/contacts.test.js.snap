// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app runs 1`] = `
<div>
  <h1>
    Contacts
  </h1>
  <h2>
    Add New
  </h2>
  <div
    className="Contact-contact-0-1-1 Contact-contactfull-0-1-2"
  >
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="firstNamefalse"
      >
        First Name: *
      </label>
      <input
        id="firstNamefalse"
        name="firstName"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="lastNamefalse"
      >
        Last Name: *
      </label>
      <input
        id="lastNamefalse"
        name="lastName"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="addressLine1false"
      >
        Address:
      </label>
      <input
        id="addressLine1false"
        name="addressLine1"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="statefalse"
      >
        State:
      </label>
      <input
        id="statefalse"
        maxLength="2"
        name="state"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="phonefalse"
      >
        Phone: *
      </label>
      <input
        id="phonefalse"
        name="phone"
        onChange={[Function]}
        required="required"
        type="tel"
        value=""
      />
    </div>
    <div
      className="Contact-requiredFields-0-1-7"
    >
      * - required fields.
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    >
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Clear"
        />
      </div>
      <div>
        <input
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Add"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`app runs 2`] = `
<div>
  <h1>
    Contacts
  </h1>
  <div
    className="Contact-contact-0-1-1 Contact-contacttiny-0-1-3 Contact-contacttinyheader-0-1-4"
  >
    <div>
      First Name
    </div>
    <div>
      LastName
    </div>
    <div>
      Phone
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    />
  </div>
  <div
    className="Contact-contact-0-1-1 Contact-contacttiny-0-1-3"
  >
    <div
      aria-label="First Name"
    >
      MyFirst
    </div>
    <div
      aria-label="Last Name"
    >
      MyLast
    </div>
    <div
      aria-label="Phone"
    >
      555-123-4567
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    >
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Delete"
        />
      </div>
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Open"
        />
      </div>
    </div>
  </div>
  <h2>
    Add New
  </h2>
  <div
    className="Contact-contact-0-1-1 Contact-contactfull-0-1-2"
  >
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="firstNamefalse"
      >
        First Name: *
      </label>
      <input
        id="firstNamefalse"
        name="firstName"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="lastNamefalse"
      >
        Last Name: *
      </label>
      <input
        id="lastNamefalse"
        name="lastName"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="addressLine1false"
      >
        Address:
      </label>
      <input
        id="addressLine1false"
        name="addressLine1"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="statefalse"
      >
        State:
      </label>
      <input
        id="statefalse"
        maxLength="2"
        name="state"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="phonefalse"
      >
        Phone: *
      </label>
      <input
        id="phonefalse"
        name="phone"
        onChange={[Function]}
        required="required"
        type="tel"
        value=""
      />
    </div>
    <div
      className="Contact-requiredFields-0-1-7"
    >
      * - required fields.
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    >
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Clear"
        />
      </div>
      <div>
        <input
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Add"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`app runs 3`] = `
<div>
  <h1>
    Contacts
  </h1>
  <div
    className="Contact-contact-0-1-1 Contact-contacttiny-0-1-3 Contact-contacttinyheader-0-1-4"
  >
    <div>
      First Name
    </div>
    <div>
      LastName
    </div>
    <div>
      Phone
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    />
  </div>
  <div
    className="Contact-contact-0-1-1 Contact-contacttiny-0-1-3"
  >
    <div
      aria-label="First Name"
    >
      MyFirst
    </div>
    <div
      aria-label="Last Name"
    >
      MyLast
    </div>
    <div
      aria-label="Phone"
    >
      555-123-4567
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    >
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Delete"
        />
      </div>
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Open"
        />
      </div>
    </div>
  </div>
  <div
    className="Contact-contact-0-1-1 Contact-contacttiny-0-1-3"
  >
    <div
      aria-label="First Name"
    >
      22222
    </div>
    <div
      aria-label="Last Name"
    >
      sssss
    </div>
    <div
      aria-label="Phone"
    >
      aaaaa
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    >
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Delete"
        />
      </div>
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Open"
        />
      </div>
    </div>
  </div>
  <h2>
    Add New
  </h2>
  <div
    className="Contact-contact-0-1-1 Contact-contactfull-0-1-2"
  >
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="firstNamefalse"
      >
        First Name: *
      </label>
      <input
        id="firstNamefalse"
        name="firstName"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="lastNamefalse"
      >
        Last Name: *
      </label>
      <input
        id="lastNamefalse"
        name="lastName"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="addressLine1false"
      >
        Address:
      </label>
      <input
        id="addressLine1false"
        name="addressLine1"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="statefalse"
      >
        State:
      </label>
      <input
        id="statefalse"
        maxLength="2"
        name="state"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="Contact-labelInput-0-1-6"
        htmlFor="phonefalse"
      >
        Phone: *
      </label>
      <input
        id="phonefalse"
        name="phone"
        onChange={[Function]}
        required="required"
        type="tel"
        value=""
      />
    </div>
    <div
      className="Contact-requiredFields-0-1-7"
    >
      * - required fields.
    </div>
    <div
      className="Contact-actionbuttons-0-1-5"
    >
      <div>
        <input
          onClick={[Function]}
          type="button"
          value="Clear"
        />
      </div>
      <div>
        <input
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Add"
        />
      </div>
    </div>
  </div>
</div>
`;
